src/herbpy/tsr/box.py

```diff
@@ -0,0 +1,45 @@
+import numpy
+import prpy.tsr
+
+@prpy.tsr.tsrlibrary.TSRFactory('herb', 'box', 'stamp')
+def box_stamp(robot, box, manip=None):
+
+    '''
+    This creates a TSR for stamping a box. 
+    It is assumed that when called, the robot is grasping a stamp
+
+    @param robot The robot to perform the stamp
+    @param box The box to stamp
+    @param manip The manipulator to stamp 
+    '''
+
+    if manip is None:
+        manip_idx = robot.GetActiveManipulatorIndex()
+    else:
+        with manip.GetRobot():
```

Comment:
Please use `CreateRobotStateSaver` to create a state saver that *only* restores the active manipulator instead of relying on the default behavior of `with robot`.